jet.py

```diff
@@ -124,7 +167,32 @@ def getP4Variations(self, df, source_dict, apply_JER=True, apply_JES=True):
         return df,source_dict
 
 
+    def getP4Variations_run3(self, df, source_dict, apply_JER, apply_JES):
+        df = df.Define("Jet_p4_shifted_map", f'''::correction::JetCorrectionProvider::getGlobal().getShiftedP4(Jet_pt, Jet_eta, Jet_phi, Jet_mass,
+                                                                                                               Jet_rawFactor, Jet_area, Rho_fixedGridRhoFastjetAll,
+                                                                                                               GenJet_pt, Jet_genJetIdx, event)''')
+
+        for source in JetCorrProducer.uncSources_core:
+            if source not in ["Central", "Total", "JER"]:
```

Comment:
This should depend whatever it was requested to apply JER, and what JES schema should be used.